dev_docs/tutorials/adding_performance_metrics.mdx

```diff
@@ -7,76 +7,156 @@ date: 2022-07-07
-- `Performance` - amount and size of events being sent
-- `Discoverability` - how easy it is to find different events, dimensions and metrics
-- `Queriability` - we would like to have the ability to write complex queries on top of telemetry data; for example getting all dashboard load events that took over 10 seconds for dashboards that contain less than 5 panels.
-- `Flexibility` - Teams should be able to flexibly design their event structure without having to define custom mappings (except for edge cases). I.e. a team can decide that the first free field is always used for that page load.
-- `Ease of visualization` - it should be easy to visualize the data
-- `Index field explosion` - having too many field names in the index, damages the performance when visualizing the data.
+The performance metrics API supports **5 numbered free fields** that can be used to report numeric metrics that you intend to analyze.
+Note that they can be used for any type of numeric information you may want to report and use to create your own flexible schema, 
+without having to add custom mappings.
 
-And event schema is composed of fields which is designed for specific purposes:
+If you want to provide event specific context, you can add properties to the `meta` field.
+The `meta` object is stored as a [flattened field](https://www.elastic.co/guide/en/elasticsearch/reference/current/flattened.html) hence 
+it's searchable and can be used to further breakdown event metrics.
 
-- `Event name` We decided to report all metric events under a single event_type, so that we can avoid registering it multiple times. This requires us to add a special standard field for eventName.
-- `Standardized fields` (duration) these fields represent common values that might be reported for performance. This can include duration for time based reports, status, heapSize, memorySize, etc. Standardized fields will be mapped as is to the index, so they can be visualized.
-- `Free fields` (key1, value1, key2, value2, ..., key5, value5) A limited set of integer-numbered free key-value pairs that can be used to report additional use-case specific fields. Teams may use them to create their own internal sub-schema. For example, a team may choose to use key1 to always report the loading time of a certain component that is always present on the page. Free fields will be mapped, if present, to the index, so they can be visualized. If a free field becomes popular, it can be promoted to a standardized field, freeing up the free field for other use.
-- `Non-standard fields` (kibana_version, protocol ...) - additional fields may be added to an event by extending them to the `meta` object. However, those won't be automatically mapped to the index. Since the `meta` field is a Flattened Field, this means those fields will be searchable by default, but an additional field mapping would be required to visualize them.
```

Comment:
I think it can be good to keep design considerations in the docs, maybe in separate section ? So we can communicate clearly with the intentions of having this schema. And we can get feedback based on these.

Reply:
We can refer to the GH issue where this was discussed maybe?
I don't want to unnecessarily duplicate this, especially as it doesn't effect the implementation.

Reply:
I am not highly opinionated on that but I'd prefer to keep things in one place (and feeling like docs is better than issues in that regards) people usually don't go through all the links and read them. But it is my opinion feel free to resolve either way ☺️